DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,113 @@
+# Developer guide
+
+This document describes how to set up a development environment for Calico, as well as how to build and test development code.
```

Comment:
Can you specifically call out that this guide shows you how to:

1. Build the images
   1. Test the images
1. Push the images to registries
1. Build manifests that reference those images